src/python/pants/notes/2.20.x.md

```diff
@@ -0,0 +1,180 @@
+
+### BUILD files
+
+The new `pants.backend.build_files.fmt.ruff` backend allows using [the Ruff formatter](https://docs.astral.sh/ruff/formatter/) to format `BUILD` files.
+
+### Remote caching/execution
+
+Pants now has experimental support for additional remote cache providers beyond the GRPC Remote Execution API. Set [the `[GLOBAL].remote_provider` option](https://www.pantsbuild.org/2.20/reference/global-options#remote_provider), to be able to use:
+
+- [the GitHub Actions Cache](https://www.pantsbuild.org/2.20/docs/using-pants/remote-caching-and-execution/remote-caching#github-actions-cache)
+- [a local directory](https://www.pantsbuild.org/2.20/docs/using-pants/remote-caching-and-execution/remote-caching#local-file-system)
+
+In support of this, [the `[GLOBAL].remote_oauth_bearer_token` option](https://www.pantsbuild.org/2.20/reference/global-options#remote_oauth_bearer_token) has been added to set the `Authorization: Bearer <token>` header. [The `[GLOBAL].remote_oauth_bearer_token_path` option](https://www.pantsbuild.org/2.20/reference/global-options#remote_oauth_bearer_token_path) has been deprecated in favour of using the new option with [a file reference](https://www.pantsbuild.org/2.20/docs/using-pants/key-concepts/options#reading-individual-option-values-from-files): `remote_oauth_bearer_token = "@/path/to/file.txt"`.
+
+[The `[GLOBAL].remote_cache_warnings` option](https://www.pantsbuild.org/2.20/reference/global-options#remote_cache_warnings) now supports `always`.
+
+### Backends
+
+#### Adhoc
+
+The [new `code_quality_tool` target](https://www.pantsbuild.org/2.20/reference/targets/code_quality_tool) in the `pants.backend.experimental.adhoc` backend allows defining "adhoc" linters, formatters and fixers without requiring a full plugin. For example, in-repository scripts or a tool not already supported by Pants itself. See [#20135](https://github.com/pantsbuild/pants/pull/20135) for an example of how to use this.
+
+#### Docker
+
+The [new `{full_directory}` interpolation](https://www.pantsbuild.org/2.20/docs/docker/tagging-docker-images#setting-a-repository-name) in `repository` and `default_repository` options expands to the full path to the `BUILD` file that contains a `docker_image` target.
+
+The new `pants.backend.experimental.docker.podman` backend allows using [Podman](https://podman.io). Add that backend to `backend_packages` and the Docker backend will invoke Podman instead of Docker. [The new option `[docker].experimental_enable_podman`](https://www.pantsbuild.org/2.20/reference/subsystems/docker#experimental_enable_podman) allows disabling this once the backend is loaded.
+
+Dependency inference now works for parameterized targets, like `ARG base="path/to:target@param=value"`.
+
+Pants is now able to extract the image ID when building a docker image with a daemon using the containerd-snapshotter feature.
+
+The default version of Hadolint has been updated from 2.10.0 to 2.12.1-beta.
```

Comment:
Suggest add: This fixes an segmentation fault error when running on MacOS. (Upstream issue -https://github.com/hadolint/hadolint/issues/919 - there wasn't one raised for Pants)

Reply:
Sounds good, thanks